backend/src/test/relay/readBlocksInBatches.test.ts

```diff
@@ -0,0 +1,99 @@
+import * as tap from 'tap';
+
+import { TxBlock } from '../../types';
+import { setup } from './common';
+
+tap.test('Relay module - readBlocksInBatches method', async (t) => {
+  const {
+    initialLastProcessedBlock,
+    chainArchive,
+    Relay,
+    defaultRelayParams,
+    batchCountLimit,
+  } = await setup();
+
+  tap.test('should return AsyncGenerator with batch of TxBlock items and last block number', async (t) => {
+    const relay = new Relay({ ...defaultRelayParams, archive: chainArchive });
+
+    const batchesGenerator = relay['readBlocksInBatches'](initialLastProcessedBlock);
+
+    for await (const [blocks, lastBlockNumber] of batchesGenerator) {
+      t.ok(blocks.length <= batchCountLimit);
+      t.type(blocks[0].block, Buffer);
+      t.type(blocks[0].metadata, Buffer);
+      // should be greater than zero if batch has items
+      t.ok(lastBlockNumber);
+    }
```

Comment:
This doesn't check if the loop iterates at all, assertion for number of iterations would help.

It also looks from other tests that this test might be simply redundant given other tests available (asserting that `readBlocksInBatches` returns async iterator would be sufficient).